### PR TITLE
Release/0.1.0

### PR DESCRIPTION
**Pull request by paulboocock**
This is the initial release of the Amplitude Tag for GTM SS which works best with the Snowplow Client but will also work with other GTM SS clients.

A free trial of Amplitude is available at: https://analytics.amplitude.com/login/my-demo

Simply requires the Amplitude HTTP V2 API Key in the settings. Everything else should just _work_. Can be configured to capture other event properties or a subset of Snowplow entities (on either the Event Properties or User Properties object).

Leverages the following Amplitude API: https://developers.amplitude.com/docs/http-api-v2

This initial 0.1.0 release only contains a single happy path test. I'd like to get this to Google asap for inclusion into the gallery. We can build out further tests for v0.2.0.

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -0,0 +1,673 @@
...
+                ],
+                "valueValidators": [
+                  {
+                    "type": "NON_EMPTY"
+                  }
+                ]
+              }
+            ],
+            "alwaysInSummary": false,
+            "help": "Specifiy the Entity name from the GTM Event, and then key you could like to map it to or leave the mapped key blank to keep the same name. These entities will populate the Amplitude `eventProperties` object.",
```
Comment (adatzer) [inline template.tpl]: nit - `Specify`

File: template.tpl [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+              },
+              {
+                "defaultValue": "",
+                "displayName": "Amplitude Mapped Key (optional)",
+                "name": "mappedKey",
+                "type": "TEXT"
+              }
+            ],
+            "alwaysInSummary": false,
+            "help": "Specifiy the Property Key from the GTM Event, and then key you could like to map it to or leave the mapped key blank to keep the same name. You can use Key Path notation here (e.g. `x-sp-tp2.p` for a Snowplow events platform or `x-sp-contexts.com_snowplowanalytics_snowplow_web_page_1.0.id` for a Snowplow events page view id (in array index 0). These keys will populate the Amplitude `eventProperties` object."
```
Comment (adatzer) [inline template.tpl]: nit - `Specify`

File: template.tpl [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+              },
+              {
+                "defaultValue": "",
+                "displayName": "Amplitude Mapped Key (optional)",
+                "name": "mappedKey",
+                "type": "TEXT"
+              }
+            ],
+            "alwaysInSummary": false,
+            "help": "Specifiy the Property Key from the GTM Event, and then key you could like to map it to or leave the mapped key blank to keep the same name.  You can use Key Path notation here (e.g. `x-sp-tp2.p` for a Snowplow events platform or `x-sp-contexts.com_snowplowanalytics_snowplow_web_page_1.0.id` for a Snowplow events page view id (in array index 0). These keys will populate the Amplitude `userProperties` object."
```
Comment (adatzer) [inline template.tpl]: nit - `Specify` , `...and the key` instead of `then key`

File: template.tpl [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+          {
+            "value": "srv",
+            "displayValue": "Server-side"
+          },
+          {
+            "value": "app",
+            "displayValue": "General App"
+          },
+          {
+            "value": "TV",
```
Comment (adatzer) [inline template.tpl]: nit - `tv` as lowercase

File: template.tpl [hunk 5]
```diff
@@ -0,0 +1,673 @@
...
+  platform: platform,
+  country: eventData['x-sp-geo_country'],
+  region: eventData['x-sp-geo_region'],
+  city: eventData['x-sp-geo_city'],
+  location_lat: eventData['x-sp-geo_latitude'],
+  location_lng: eventData['x-sp-geo_longitude'],
+  carrier: eventData['x-sp-ip_organization'],
+  language: eventData.language,
+  //"event_id": 23, // Unable to populate
+  //"session_id": 1396381378123, // Unable to populate
```
Comment (adatzer) [inline template.tpl]: Can the commented lines be removed or would it make sense to add some explanation on the comments?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,92 @@
...
+[docs]: https://docs.snowplowanalytics.com
+[snowplow]: https://github.com/snowplow/snowplow
+[discourse]: https://discourse.snowplowanalytics.com
+
+[techdocs]: https://docs.snowplowanalytics.com/docs/
+[techdocs-image]: https://d3i6fms1cm1j0i.cloudfront.net/github/images/techdocs.png
+[setup]: https://docs.snowplowanalytics.com/docs/
+[setup-image]: https://d3i6fms1cm1j0i.cloudfront.net/github/images/setup.png
+
+[contributing]: https://github.com/snowplow/snowplow-gtm-server-side-amplitude-tag/blob/master/CONTRIBUTING.md
```
Comment (adatzer) [inline README.md]: Could this be changed to `https://github.com/snowplow/snowplow-gtm-server-side-amplitude-tag/blob/main/CONTRIBUTING.md` (or just to `CONTRIBUTING.md`)?

File: template.tpl [hunk 7]
```diff
@@ -0,0 +1,673 @@
...
+  }
+]
+
+
+___SANDBOXED_JS_FOR_SERVER___
+
+const getEventData = require('getEventData');
+const sendHttpRequest = require('sendHttpRequest');
+const JSON = require('JSON');
+const log = require('logToConsole');
```
Comment (adatzer) [inline template.tpl]: Maybe could `log` be removed, since not used?

**Review by adatzer — APPROVED** [review]
Looks great! With many helpers for other tags as well!
Just added few nit comments.

File: template.tpl (hunk 1 above)
Comment (paulboocock) [reply template.tpl]: Why can't I type Specify? 😭

**Pull request merged**